imagecodecs/_jpeg8.pyx

```diff
@@ -152,8 +152,9 @@ def jpeg8_encode(
 
     if bitspersample is not None:
         if (
-            bitspersample not in {8, 12, 16}
-            or src.itemsize == 1 and bitspersample > 8
+            not (lossless and 2 <= bitspersample <= 16)
+            or (not lossless and bitspersample not in {8, 12})
+            or src.itemsize < (bitspersample + 7) // 8
```

Comment:
Should this be `!=` instead of `<`? Or how should the case with bitspersample <= 8 stored in uint16 arrays be handled?